src/JWT.ts

```diff
@@ -244,8 +251,40 @@ export function decodeJWT(jwt: string): JWTDecoded {
   if (!jwt) throw new Error('invalid_argument: no JWT passed into decodeJWT')
   try {
     const jws = decodeJWS(jwt)
-    const decodedJwt: JWTDecoded = Object.assign(jws, { payload: JSON.parse(decodeBase64url(jws.payload)) })
-    return decodedJwt
+    if (jws.header.cty === "JWT") {
+      // This is a multi-signature JWT using the nested JWT structure
+      // See point 5 of https://www.rfc-editor.org/rfc/rfc7519#section-7.1
+      const signature: GeneralJWSSignature[] = [];
+      const data = jws.data;
+      let i = 0;
+
+      let loopJws: JWSDecoded = {} as any;
+      Object.assign(loopJws, jws)
+
+      while (i < 200 && loopJws.header.cty === "JWT") {
+        signature.push({
+          protected: loopJws.header,
+          signature: loopJws.signature,
+        })
+        let payload = decodeBase64url(loopJws.payload);
+        loopJws = decodeJWS(payload)
```

Comment:
will this not mess the value inside the signature array as objects in javascript are linked

Reply:
this overwrites the value of the object (a pointer) each time it does re-assignment. So the old value is discarded (garbage collector deletes it I assume), and new value is used.